pkg/controllers/managementagent/monitoring/operatorHandler.go

```diff
@@ -40,32 +40,37 @@ func (h *operatorHandler) syncCluster(key string, obj *mgmtv3.Cluster) (runtime.
 		return obj, nil
 	}
 
-	var newCluster *mgmtv3.Cluster
-	var err error
+	cpy := obj.DeepCopy()
+	var err, updateErr error
 	//should deploy
 	if obj.Spec.EnableClusterAlerting || obj.Spec.EnableClusterMonitoring {
-		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(obj, func() (runtime.Object, error) {
-			cpy := obj.DeepCopy()
+		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(cpy, func() (runtime.Object, error) {
 			return cpy, deploySystemMonitor(cpy, h.app, h.catalogManager, h.clusterName)
 		})
 		if err != nil {
 			logrus.Warnf("deploy prometheus operator error, %v", err)
```

Comment:
We should not be logging this or the one below, the contents of the log is what should be returned at the end in the error.